### PR TITLE
ci: added github pipelines and removed gitlab pipelines

### PR DESCRIPTION
**Pull request by rH4rtinger**

### CODE REVIEW

**Review by SKreileder — DISMISSED** [review]

File: .github/workflows/nodejs.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+name: Build Project
+
+on:
+  pull_request:
```
Comment (wglanzer) [inline .github/workflows/nodejs.yml:4]: Is this really the "right" and common way in public repositories?
I guess anyone is able to run our pipelines now, without any approval or something.

File: .github/workflows/npm-publish.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-node@v4
+        with:
+          node-version: 20
+          registry-url: https://registry.npmjs.org/
+      - run: npm ci
+      - run: npm publish --access public
+        env:
+          NODE_AUTH_TOKEN: ${{secrets.NPM_TOKEN}}
```
Comment (wglanzer) [inline .github/workflows/npm-publish.yml:32]: I think this should be removed?
It seems that secrets get removed on console output, yes, but if I write the secret in a file, like `echo ${{secrets.NPM_TOKEN}} > ./mySecret`, and after that I export it as an artifact, then the secret is completely visible to anyone... Isn't it?

**Review by wglanzer — CHANGES_REQUESTED** [review]

File: .github/workflows/nodejs.yml (hunk 1 above)
Comment (rH4rtinger) [reply .github/workflows/nodejs.yml:4]: Firstly, CI pipelines are run at the branch were they are defined. [Just running those from the main branch](https://github.com/orgs/community/discussions/53430) is not supported at the moment.

There would be a [Push ruleset](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/managing-rulesets/about-rulesets#push-rulesets), but these are not available for public repos:
> Push rulesets are available for the GitHub Team plan in internal and private repositories, and forks of repositories that have push rulesets enabled.

I've set the action settings to the following:

* **Action Permissions:** Allow aditosoftware actions and reusable workflows. Any action or reusable workflow defined in a repository within aditosoftware can be used.
* Fork pull request workflows from outside collaborators: Require approval for all outside collaborators

This should put us in control of running the workflows when changes from the forks are coming.



----


For running workflows in pull requests, I looked at the  [example node.js workflow](https://github.com/aditosoftware/eslint-config-adito-platform/new/main?filename=.github%2Fworkflows%2Fnode.js.yml&workflow_template=ci%2Fnode.js)

File: .github/workflows/npm-publish.yml (hunk 2 above)
Comment (rH4rtinger) [reply .github/workflows/npm-publish.yml:32]: See https://github.com/aditosoftware/eslint-config-adito-platform/pull/1#discussion_r1754083657 and  [the example node.js publish pipeline](https://github.com/aditosoftware/eslint-config-adito-platform/new/main?filename=.github%2Fworkflows%2Fnpm-publish.yml&workflow_template=ci%2Fnpm-publish)

Comment (wglanzer) [reply .github/workflows/npm-publish.yml:32]: So we have to have an eye on contributors, that change our workflows in pullrequests, haven't we?

Comment (rH4rtinger) [reply .github/workflows/npm-publish.yml:32]: Yes we do have. 
We need to check the commits and see their changed workflows and only if we think they are good, then we are running those.

**Review by wglanzer — APPROVED** [review]

**Pull request merged**